src/app/search/page.tsx

```diff
@@ -1,17 +1,153 @@
+      pb={4}
+    >
+      <Text textStyle={'text-regular-sm'} color={'textSecondary'}>
+        Search results for:
+      </Text>
+      {Object.entries(filters).map(([key, value]) => {
+        if (!value || (Array.isArray(value) && value.length === 0)) return null;
+        const keyword = getKeywordByFilter(key);
+        const displayValue = Array.isArray(value) ? value.join(', ') : value;
+        return (
+          <Text key={key} textStyle={'text-regular-sm'} color={'textPrimary'}>
+            {!!keyword && (
+              <Text
+                textStyle={'text-medium-sm'}
+                fontWeight={'medium'}
+                display="inline-block"
+                bg={{ base: 'feedback.blue-200', _dark: 'feedback.blue-900' }}
+                color={{ base: 'feedback.blue-600', _dark: 'feedback.blue-300' }}
+                borderRadius="redesign.xs"
+                py={1}
+                whiteSpace="pre"
+                textTransform={'uppercase'}
+              >
+                {' '}
+                {keyword}{' '}
+              </Text>
+            )}{' '}
+            {displayValue && filterToFormattedValueMap[key]
+              ? filterToFormattedValueMap[key](displayValue)
+              : displayValue || ''}
+          </Text>
+        );
+      })}
+    </Flex>
   );
+}
+
+function SearchContent({ onTotalChange }: { onTotalChange: (total: number) => void }) {
+  return (
+    <Stack gap={5} mt={5}>
+      <TxTableFilters />
+      <TxSearchResult onTotalChange={onTotalChange} />
+      <TxTableFiltersModal />
+    </Stack>
+  );
+}
 
-  return <Page filters={filters} />;
+function SearchPageContent() {
+  const { fromAddress, toAddress, startTime, endTime, transactionType } = useTxTableFilters();
+  const [resultCount, setResultCount] = useState<number | null>(null);
+
+  const filters = {
+    fromAddress,
+    toAddress,
+    startTime,
+    endTime,
+    transactionType,
+  };
+
+  return (
+    <Stack gap={24} fontFamily="var(--font-instrument-sans)">
+      <Stack gap={3}>
+        <SearchHeader resultCount={resultCount} />
+        <SearchFiltersDisplay filters={filters} />
+        <SearchContent onTotalChange={setResultCount} />
+      </Stack>
+    </Stack>
+  );
+}
+
+export default function SearchPage() {
+  const searchParams = useSearchParams();
+
+  const defaultFromAddress = searchParams.get('fromAddress') || '';
+  const defaultToAddress = searchParams.get('toAddress') || '';
+  const defaultStartTime = searchParams.get('startTime') || '';
+  const defaultEndTime = searchParams.get('endTime') || '';
+  const defaultTransactionType = searchParams.get('transactionType')?.split(',') || [];
+
+  return (
+    <TxTableFiltersProvider
+      defaultFromAddress={defaultFromAddress}
+      defaultToAddress={defaultToAddress}
+      defaultStartTime={defaultStartTime}
+      defaultEndTime={defaultEndTime}
+      defaultTransactionType={defaultTransactionType}
+    >
+      <SearchPageContent />
```

Comment:
I would put all of the components related to searchpagecontent into a file called searchpage content to this file clean and easy to read